flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/RuntimeFilterITCase.java

```diff
@@ -0,0 +1,213 @@
+        tEnv.getConfig().set(OptimizerConfigOptions.TABLE_OPTIMIZER_RUNTIME_FILTER_ENABLED, true);
+        tEnv.getConfig()
+                .set(
+                        OptimizerConfigOptions.TABLE_OPTIMIZER_RUNTIME_FILTER_MAX_BUILD_DATA_SIZE,
+                        MemorySize.parse("10m"));
+        tEnv.getConfig()
+                .set(
+                        OptimizerConfigOptions.TABLE_OPTIMIZER_RUNTIME_FILTER_MIN_PROBE_DATA_SIZE,
+                        MemorySize.parse("10g"));
+        tEnv.getConfig()
+                .set(OptimizerConfigOptions.TABLE_OPTIMIZER_RUNTIME_FILTER_MIN_FILTER_RATIO, 0.5);
+
+        tEnv.executeSql(
+                String.format(
+                        "CREATE TABLE dim (\n"
+                                + "  x INT,\n"
+                                + "  y INT,\n"
+                                + "  z BIGINT\n"
+                                + ")  WITH (\n"
+                                + " 'connector' = 'values',\n"
+                                + " 'disable-lookup' = 'true',\n"
+                                + " 'data-id' = '%s',\n"
+                                + " 'bounded' = 'true'\n"
+                                + ")",
+                        TestValuesTableFactory.registerData(TestData.data7())));
+
+        tEnv.executeSql(
+                String.format(
+                        "CREATE TABLE fact (\n"
+                                + "  `a` INT,\n"
+                                + "  `b` BIGINT,\n"
+                                + "  `c` INT,\n"
+                                + "  `d` VARCHAR,\n"
+                                + "  `e` BIGINT\n"
+                                + ") WITH (\n"
+                                + "  'connector' = 'values',\n"
+                                + "  'runtime-source' = 'NewSource',\n"
+                                + "  'data-id' = '%s',\n"
+                                + "  'disable-lookup' = 'true',\n"
+                                + "  'bounded' = 'true'\n"
+                                + ")",
+                        TestValuesTableFactory.registerData(TestData.data5())));
+
+        // note that we set the row count here so that the runtime filter can take effect
+        // it is not the real data volume of the table
+        catalog.alterTableStatistics(
+                new ObjectPath(tEnv.getCurrentDatabase(), "dim"),
+                new CatalogTableStatistics(
+                        FlinkRuntimeFilterProgramTest.SUITABLE_DIM_ROW_COUNT, 1, 1, 1),
+                false);
+        catalog.alterTableStatistics(
+                new ObjectPath(tEnv.getCurrentDatabase(), "fact"),
+                new CatalogTableStatistics(
+                        FlinkRuntimeFilterProgramTest.SUITABLE_FACT_ROW_COUNT, 1, 1, 1),
+                false);
+    }
+
+    @ParameterizedTest(name = "mode = {0}")
+    @MethodSource("parameters")
+    void testSimpleRuntimeFilter(BatchShuffleMode shuffleMode) {
```

Comment:
Do you have check the plan whether is as expected?

Reply:
Yes, have checked.